frontend/class-breadcrumbs.php

```diff
@@ -664,10 +664,19 @@ private function prepare_links() {
 	 * @return array Array of link text and url
 	 */
 	private function get_link_info_for_id( $id ) {
-		$link = array();
 
+		$link = array();
 		$link['url']  = get_permalink( $id );
+		$status = get_post_status( $id );
+
+		if ( 'private' == $status && ! current_user_can( 'read_private_pages' ) && ! current_user_can( 'read_private_posts' ) ) {
```

Comment:
String comparison should be strict. Also I would expect capability check to match the post type.